protobuf-provider/src/main/java/io/confluent/kafka/schemaregistry/protobuf/ProtobufSchemaUtils.java

```diff
@@ -113,7 +114,14 @@ public static String toString(ProtoFileElement protoFile) {
     if (!protoFile.getTypes().isEmpty()) {
       sb.append('\n');
       for (TypeElement typeElement : protoFile.getTypes()) {
-        sb.append(toString(typeElement));
+        if (typeElement instanceof MessageElement) {
+          sb.append(toString((MessageElement) typeElement));
+        }
+      }
+      for (TypeElement typeElement : protoFile.getTypes()) {
+        if (typeElement instanceof EnumElement) {
+          sb.append(toString((EnumElement) typeElement));
+        }
```

Comment:
Can the `typeElement` here be something else than `MessageElement` and `EnumElement`? In that case it won't be appended to the canonical string

Reply:
Good question.  No, those are the only two subtypes.